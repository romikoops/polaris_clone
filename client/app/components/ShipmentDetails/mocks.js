// Will remove it later
// ============================================
import { get } from 'lodash'

export const trucking = { preCarriage: { truckType: '' }, onCarriage: { truckType: '' } }
export const id = 4606
export const selectedDay = '2019-01-24T10:00:00.000Z'

export const ShipmentDetailsAvailableRoutes = [{
  itineraryId: 2863,
  itineraryName: 'Gothenburg - Qingdao',
  modeOfTransport: 'ocean',
  origin: {
    stopId: 4656, hubId: 3023, hubName: 'Gothenburg Port', nexusId: 597, nexusName: 'Gothenburg', latitude: 57.694253, longitude: 11.854048, country: 'SE', truckTypes: ['default']
  },
  destination: {
    stopId: 4657, hubId: 3027, hubName: 'Qingdao Port', nexusId: 601, nexusName: 'Qingdao', latitude: 36.083811, longitude: 120.323534, country: 'CN', truckTypes: []
  }
}, {
  itineraryId: 2958,
  itineraryName: 'Gothenburg - Qingdao',
  modeOfTransport: 'air',
  origin: {
    stopId: 4846, hubId: 3030, hubName: 'Gothenburg Airport', nexusId: 597, nexusName: 'Gothenburg', latitude: 57.694253, longitude: 11.854048, country: 'SE', truckTypes: ['default']
  },
  destination: {
    stopId: 4847, hubId: 3034, hubName: 'Qingdao Airport', nexusId: 601, nexusName: 'Qingdao', latitude: 36.083811, longitude: 120.323534, country: 'CN', truckTypes: []
  }
}]

export const ShipmentDetails = {
  availableRoutes: ShipmentDetailsAvailableRoutes,
  availableMots: ['ocean', 'air']
}

export const cargoItemTypes = [{
  id: 25, dimension_x: null, dimension_y: null, description: 'Pallet', area: null, created_at: '2018-06-27T17:28:28.431Z', updated_at: '2018-06-27T17:28:28.431Z', category: 'Pallet'
}, {
  id: 22, dimension_x: null, dimension_y: null, description: 'Carton', area: null, created_at: '2018-06-27T17:28:28.417Z', updated_at: '2018-06-27T17:28:28.417Z', category: 'Carton'
}, {
  id: 23, dimension_x: null, dimension_y: null, description: 'Crate', area: null, created_at: '2018-06-27T17:28:28.422Z', updated_at: '2018-06-27T17:28:28.422Z', category: 'Crate'
}, {
  id: 26, dimension_x: null, dimension_y: null, description: 'Bottle', area: null, created_at: '2018-06-27T17:28:28.436Z', updated_at: '2018-06-27T17:28:28.436Z', category: 'Bottle'
}, {
  id: 27, dimension_x: null, dimension_y: null, description: 'Stack', area: null, created_at: '2018-06-27T17:28:28.440Z', updated_at: '2018-06-27T17:28:28.440Z', category: 'Stack'
}, {
  id: 28, dimension_x: null, dimension_y: null, description: 'Drum', area: null, created_at: '2018-06-27T17:28:28.445Z', updated_at: '2018-06-27T17:28:28.445Z', category: 'Drum'
}, {
  id: 29, dimension_x: null, dimension_y: null, description: 'Skid', area: null, created_at: '2018-06-27T17:28:28.450Z', updated_at: '2018-06-27T17:28:28.450Z', category: 'Skid'
}, {
  id: 30, dimension_x: null, dimension_y: null, description: 'Barrel', area: null, created_at: '2018-06-27T17:28:28.454Z', updated_at: '2018-06-27T17:28:28.454Z', category: 'Barrel'
}]

export const maxDimensions = {
  general: {
    dimensionX: '590.0', dimensionY: '234.2', dimensionZ: '228.0', payloadInKg: '21770.0', chargeableWeight: '21770.0'
  },
  air: {
    dimensionX: '120.0', dimensionY: '100.0', dimensionZ: '150.0', payloadInKg: '1000.0', chargeableWeight: '1000.0'
  }
}

export const scope = {
  links: { about: '', legal: '' },
  terms: ['You verify that all the information provided above is true', 'You agree to the presented terms and conditions.', 'Demo is to discuss the validity of the presented prices with the product owners.'],
  fee_detail: 'key_and_name',
  closed_shop: false,
  has_customs: true,
  has_insurance: true,
  fixed_currency: false,
  dangerous_goods: false,
  cargo_info_level: 'hs_codes',
  carriage_options: { on_carriage: { export: 'optional', import: 'optional' }, pre_carriage: { export: 'optional', import: 'optional' } },
  detailed_billing: false,
  total_dimensions: true,
  consolidate_cargo: false,
  modes_of_transport: {
    air: { container: true, cargo_item: true }, rail: { container: true, cargo_item: true }, ocean: { container: true, cargo_item: true }, truck: { container: false, cargo_item: false }
  },
  show_beta_features: true,
  closed_registration: false,
  continuous_rounding: false,
  incoterm_info_level: 'text',
  non_stackable_goods: true,
  open_quotation_tool: false,
  customs_export_paper: false,
  fixed_exchange_rates: false,
  require_full_address: true,
  closed_quotation_tool: false
}

export const theme = {
  colors: {
    primary: '#0D5BA9', secondary: '#23802A', brightPrimary: '#2491FD', brightSecondary: '#25ED36'
  },
  logoWide: 'https://assets.itsmycargo.com/assets/logos/Logo_transparent_blue.png',
  logoLarge: 'https://assets.itsmycargo.com/assets/logos/logo_box.png',
  logoSmall: 'https://assets.itsmycargo.com/assets/logos/logo_box.png',
  logoWhite: 'https://assets.itsmycargo.com/assets/logos/Logo_transparent_white.png',
  background: 'https://assets.itsmycargo.com/assets/images/cropped_banner_2.jpg'
}

export const origin = {
  latitude: 57.694253, longitude: 11.854048, nexusId: 597, nexusName: 'Gothenburg', country: 'SE'
}
export const destination = {
  latitude: 36.083811, longitude: 120.323534, nexusId: 601, nexusName: 'Qingdao', country: 'CN'
}
export const cargoItem = {
  payloadInKg: 11, totalVolume: 0, totalWeight: 0, dimensionX: 7, dimensionY: 7, dimensionZ: 8, quantity: 7, cargoItemTypeId: 23, dangerousGoods: false, stackable: false
}

export const cargoItemAggregated = {
  payloadInKg: 0, totalVolume: 122, totalWeight: 346, dimensionX: 0, dimensionY: 0, dimensionZ: 0, quantity: 1, cargoItemTypeId: '', dangerousGoods: false, stackable: true
}

export const cargoItemContainer = {
  sizeClass: 'highCube', quantity: 14, dangerousGoods: false, weight: 16
}

export const cargoUnits = [cargoItem]

export const identity = x => x

export const importedProps = {
  ShipmentDetails,
  aggregatedCargo: false,
  cargoItemTypes,
  cargoUnits,
  destination,
  direction: 'export',
  id,
  loadType: 'cargo_item',
  maxDimensions,
  onCarriage: false,
  onChangeCargoUnitCheckbox: identity,
  onChangeCargoUnitInput: identity,
  onChangeCargoUnitSelect: identity,
  onDeleteUnit: identity,
  onUpdateCargoUnit: identity,
  origin,
  preCarriage: false,
  scope,
  selectedDay,
  theme,
  toggleModal: identity,
  trucking
}

export function logPath (props, path) {
  console.log(path, get(props, path, 'NO SUCH PATH'))
}

export function log (props) {
  let holder = '{\n'
  const declarations = []
  Object.keys(props).forEach((key) => {
    const toLog = `const ${key} = ${JSON.stringify(props[key])}`
    if (toLog.length < 150) console.log(toLog)
    declarations.push(toLog)

    holder = `${holder}\n  ${key},`
  })
  holder = `${holder}\n}`
  console.log(declarations.join('\n'))
  // console.log(holder)
  // console.log(JSON.stringify(props, null, 2))
}
