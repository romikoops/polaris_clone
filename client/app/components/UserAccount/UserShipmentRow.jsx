import React, { Component } from 'react'
import { v4 } from 'uuid'
import PropTypes from '../../prop-types'
import styles from './UserShipmentRow.scss'
import { moment } from '../../constants'

export class UserShipmentRow extends Component {
  static switchIcon (sched) {
    let icon
    switch (sched.mode_of_transport) {
      case 'ocean':
        icon = <i className="fa fa-ship" />
        break
      case 'air':
        icon = <i className="fa fa-plane" />
        break
      case 'train':
        icon = <i className="fa fa-train" />
        break
      default:
        icon = <i className="fa fa-ship" />
        break
    }
    return icon
  }
  static calcCargoLoad (feeHash, loadType) {
    const cargoCount = Object.keys(feeHash.cargo).length
    let noun = ''
    if (loadType === 'cargo_item' && cargoCount > 1) {
      noun = 'Cargo Items'
    } else if (loadType === 'cargo_item' && cargoCount === 1) {
      noun = 'Cargo Item'
    } else if (loadType === 'container' && cargoCount > 1) {
      noun = 'Containers'
    } else if (loadType === 'container' && cargoCount === 1) {
      noun = 'Container'
    }
    return `${cargoCount} X ${noun}`
  }

  static dashedGradient (color1, color2) {
    return `linear-gradient(to right, transparent 70%, white 30%), linear-gradient(to right, ${color1}, ${color2})`
  }

  constructor (props) {
    super(props)
    this.selectShipment = this.selectShipment.bind(this)
    this.handleDeny = this.handleDeny.bind(this)
    this.handleAccept = this.handleAccept.bind(this)
  }
  componentDidMount () {
    window.scrollTo(0, 0)
  }
  selectShipment () {
    const { shipment, handleSelect } = this.props
    handleSelect(shipment)
  }
  handleDeny () {
    const { shipment, handleAction } = this.props
    handleAction(shipment.id, 'decline')
  }

  handleAccept () {
    const { shipment, handleAction } = this.props
    handleAction(shipment.id, 'accept')
  }
  render () {
    const {
      theme, shipment, hubs, user
    } = this.props
    const schedule = {}
    const originHub = hubs[shipment.origin_hub_id].data
    const destHub = hubs[shipment.destination_hub_id].data
    const gradientFontStyle = {
      background:
        theme && theme.colors
          ? `-webkit-linear-gradient(left, ${theme.colors.brightPrimary}, ${
            theme.colors.brightSecondary
          })`
          : 'black'
    }
    const dashedLineStyles = {
      marginTop: '6px',
      height: '2px',
      width: '100%',
      background:
        theme && theme.colors
          ? UserShipmentRow.dashedGradient(theme.colors.primary, theme.colors.secondary)
          : 'black',
      backgroundSize: '16px 2px, 100% 2px'
    }
    // const pendingRow = (
    //   <div className="flex-50 layout-row layout-align-end-end layout-wrap">
    //     <div className="flex-none layout-row layout-align-start-end">
    //       <p className="flex-none">Status: </p>
    //     </div>
    //     <div className={`flex-40 layout-row layout-align-center-end ${styles.pending}`}>
    //       <i className="flex-none fa fa-clock-o" />
    //       <p className="flex-none">Pending</p>
    //     </div>
    //   </div>
    // )
    // const acceptedRow = (
    //   <div className="flex-50 layout-row layout-align-end-end layout-wrap">
    //     <div className="flex-none layout-row layout-align-start-end">
    //       <p className="flex-none">Status: </p>
    //     </div>
    //     <div className={`flex-40 layout-row layout-align-center-end ${styles.grant}`}>
    //       <i className="flex-none fa fa-check" />
    //       <p className="flex-none">Accepted</p>
    //     </div>
    //   </div>
    // )
    // const deniedRow = (
    //   <div className="flex-50 layout-row layout-align-end-end layout-wrap">
    //     <div className="flex-none layout-row layout-align-start-end">
    //       <p className="flex-none">Status: </p>
    //     </div>
    //     <div className={`flex-40 layout-row layout-align-center-end ${styles.deny}`}>
    //       <i className="flex-none fa fa-trash" />
    //       <p className="flex-none">Denied</p>
    //     </div>
    //   </div>
    // )
    const feeHash = shipment.selected_offer
    return (
      <div key={v4()} className={`flex-100 layout-row pointy ${styles.route_result}`}>
        <div
          className={`${styles.port_box} flex-25 layout-row layout-wrap`}
          onClick={this.selectShipment}
        >
          <div className="flex-100 layout-row layout-align-center-center">
            <h5 className="flex-none no_m">
              {' '}
              {`Booking Placed at: ${moment(shipment.booking_placed_at).format('DD/MM/YYYY HH:mm')}`}
            </h5>
          </div>
          <div className={`${styles.hub_half} flex-100 layout-row layout-align-center-center`}>
            <div className="flex-10 layout-row layout-align-center-center">
              <i className={`flex-none fa fa-map-marker ${styles.map_marker}`} />
            </div>
            <div className="flex layout-row layout-align-center-center">
              <h4 className="flex-100 center letter_3"> {originHub.name} </h4>
            </div>
          </div>
          <div
            className={`${
              styles.hub_mid_line
            } flex-none layout-row layout-align-space-around-center layout-wrap`}
          >
            {/* <i className="fa fa-chevron-down flex-none" />
            {/* <i className="fa fa-chevron-down flex-none" /> */}

            {/* <i className="fa fa-chevron-down flex-none" /> */}
            {/* <i className="fa fa-chevron-down flex-none" /> */}
            <div className="flex-100 layout-row layout-align-center-end">
              {UserShipmentRow.switchIcon(schedule, gradientFontStyle)}
            </div>
            <div style={dashedLineStyles} />
          </div>
          <div className={`${styles.hub_half} flex-100 layout-row layout-align-center-center`}>
            <div className="flex-10 layout-row layout-align-center-center">
              <i className={`flex-none fa fa-flag-o ${styles.flag}`} />
            </div>
            <div className="flex layout-row layout-align-center-center">
              <h4 className="flex-100 center letter_3"> {destHub.name} </h4>
            </div>
          </div>
        </div>
        <div
          className={`${styles.main_info} flex layout-row layout-align-none-start layout-wrap`}
          onClick={this.selectShipment}
        >
          <div className="flex-100 layout-row layout-align-none-center">
            <div className="flex-50 layout-row layout-align-start-center">
              <h4 className="flex-none no_m letter_3"> {`Ref: ${shipment.imc_reference}`}</h4>
            </div>

            <div className="flex-50 layout-row layout-align-end-center">
              <h4 className="flex-none letter_3 no_m">
                {' '}
                {`${UserShipmentRow.calcCargoLoad(feeHash, shipment.load_type)} `}
              </h4>
            </div>
          </div>
          <div className="flex-100 layout-row layout-align-none-center">
            <div className={`${styles.user_info} flex-50 layout-row layout-align-start-center`}>
              <i className={`flex-none fa fa-user ${styles.flag}`} style={gradientFontStyle} />
              <h4 className="flex-none letter_3"> {`${user.first_name} ${user.last_name}`} </h4>
            </div>
            <div className="flex-50 layout-row layout-align-end-center">
              <h4 className="flex-none letter_3 no_m">
                {' '}
                {`Total: ${shipment.total_price.currency} ${parseFloat(shipment.total_price.value).toFixed(2)}`}
              </h4>
            </div>
          </div>

          <div className="flex-100 layout-row layout-align-none-center">
            <div className="flex-25 layout-wrap layout-row layout-align-center-center">
              <div className="flex-100 layout-row">
                <h4 className={styles.date_title}>
                  {shipment.has_pre_carriage ? 'Pickup Date' : 'Closing Date'}
                </h4>
              </div>
              <div className="flex-100 layout-row">
                <p className={`flex-none ${styles.sched_elem}`}>
                  {' '}
                  {moment(shipment.planned_pickup_date).format('DD/MM/YYYY')}{' '}
                </p>
                <p className={`flex-none ${styles.sched_elem}`}>
                  {' '}
                  {moment(shipment.planned_pickup_date).format('HH:mm')}{' '}
                </p>
              </div>
            </div>
            <div className="flex-25 layout-wrap layout-row layout-align-center-center">
              <div className="flex-100 layout-row">
                <h4 className={styles.date_title}> ETD</h4>
              </div>
              <div className="flex-100 layout-row">
                <p className={`flex-none ${styles.sched_elem}`}>
                  {' '}
                  {moment(shipment.planned_etd).format('DD/MM/YYYY')}{' '}
                </p>
                <p className={`flex-none ${styles.sched_elem}`}>
                  {' '}
                  {moment(shipment.planned_etd).format('HH:mm')}{' '}
                </p>
              </div>
            </div>
            <div className="flex-25 layout-wrap layout-row layout-align-center-center">
              <div className="flex-100 layout-row">
                <h4 className={styles.date_title}> ETA </h4>
              </div>
              <div className="flex-100 layout-row">
                <p className={`flex-none ${styles.sched_elem}`}>
                  {' '}
                  {moment(shipment.planned_eta).format('DD/MM/YYYY')}{' '}
                </p>
                <p className={`flex-none ${styles.sched_elem}`}>
                  {' '}
                  {moment(shipment.planned_eta).format('HH:mm')}{' '}
                </p>
              </div>
            </div>
            <div className="flex-25 layout-wrap layout-row layout-align-center-center">
              <div className="flex-100 layout-row">
                <h4 className={styles.date_title}> Estimated Transit Time </h4>
              </div>
              <div className="flex-100 layout-row">
                <p className={`flex-none ${styles.sched_elem}`}>
                  {' '}
                  {moment(shipment.planned_eta).diff(shipment.planned_etd, 'days')}
                  {'  Days'}
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    )
  }
}
UserShipmentRow.propTypes = {
  theme: PropTypes.theme,
  handleSelect: PropTypes.func.isRequired,
  handleAction: PropTypes.func.isRequired,
  hubs: PropTypes.arrayOf(PropTypes.object),
  shipment: PropTypes.shipment.isRequired,
  user: PropTypes.objectOf(PropTypes.String).isRequired
}

UserShipmentRow.defaultProps = {
  theme: null,
  hubs: []
}

export default UserShipmentRow
