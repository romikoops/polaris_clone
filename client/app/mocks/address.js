export const firstAddress = {
  user: {
    primary: true
  },
  primary: true,
  id: 246,
  streetNumber: 'FOO_ADDRESS_STREET_NUMBER',
  geocodedAddress: 'FOO_GEOCODED_ADDRESS',
  street: 'FOO_ADDRESS_STREET',
  city: 'FOO_ADDRESS_CITY',
  zipCode: 'FOO_ADDRESS_ZIP_CODE',
  country: 'FOO_ADDRESS_COUNTRY'
}

export const secondAddress = {
  user: {
    primary: false
  },
  primary: false,
  id: 704,
  streetNumber: 'BAR_ADDRESS_STREET_NUMBER',
  geocodedAddress: 'BAR_GEOCODED_ADDRESS',
  street: 'BAR_ADDRESS_STREET',
  city: 'BAR_ADDRESS_CITY',
  zipCode: 'BAR_ADDRESS_ZIP_CODE',
  country: 'BAR_ADDRESS_COUNTRY'
}

export const originAddress = {
  city: 'ORIGIN_ADDRESS_CITY',
  country: 'ORIGIN_ADDRESS_COUNTRY',
  street: 'ORIGIN_ADDRESS_STREET',
  street_number: 'ORIGIN_ADDRESS_STREET_NUMBER',
  zip_code: 'ORIGIN_ADDRESS_ZIP_CODE'
}
export const destinationAddress = {
  city: 'DESTINATION_ADDRESS_CITY',
  country: 'DESTINATION_ADDRESS_COUNTRY',
  street: 'DESTINATION_ADDRESS_STREET',
  street_number: 'DESTINATION_ADDRESS_STREET_NUMBER',
  zip_code: 'DESTINATION_ADDRESS_ZIP_CODE'
}

export const addresses = {
  origin: originAddress,
  destination: destinationAddress
}
