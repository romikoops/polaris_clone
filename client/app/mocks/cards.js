export const firstCard = {
  name: 'FOO_NAME',
  img: 'FOO_IMG',
  url: 'FOO_URL',
  handleClick: x => x,
  options: {
    name: 'FOO_OPTIONS_NAME'
  }
}
export const secondCard = {
  name: 'BAR_NAME',
  img: 'BAR_IMG',
  url: 'BAR_URL',
  handleClick: x => x,
  options: {
    name: 'BAR_OPTIONS_NAME'
  }
}
export const cards = [
  firstCard,
  secondCard
]
