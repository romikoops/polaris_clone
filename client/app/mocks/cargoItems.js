export const firstCargoItem = {
  cargo_item_type_id: 'foo',
  chargeable_weight: 250,
  dangerous_goods: false,
  description: 'FOO_CARGO_ITEM_DESCRIPTION',
  dimension_x: 10,
  dimension_y: 60,
  dimension_z: 40,
  gross_weight: 76,
  hs_codes: [4],
  hs_text: 'FOO_CARGO_ITEM_HS_TEXT',
  id: 1,
  key: 'FOO_CARGO_ITEM_KEY',
  payload_in_kg: 200,
  quantity: 5,
  stackable: true,
  tare_weight: 20,
  weight: 220
}

export const secondCargoItem = {
  cargo_item_type_id: 'bar',
  chargeable_weight: 150,
  description: 'BAR_CARGO_ITEM_DESCRIPTION',
  dimension_x: 100,
  dimension_y: 50,
  dimension_z: 70,
  hs_codes: [],
  hs_text: 'BAR_CARGO_ITEM_HS_TEXT',
  id: 2,
  key: 'BAR_CARGO_ITEM_KEY',
  payload_in_kg: 100,
  quantity: 7,
  stackable: false
}

export const cargoItems = [firstCargoItem, secondCargoItem]
export const cargoItemTypes = { foo: { description: 'FOO_TYPE' }, bar: { description: 'BAR_TYPE' } }

export const cargoItemGroup = {
  dimension_y: parseFloat(firstCargoItem.dimension_y) * parseInt(firstCargoItem.quantity, 10),
  dimension_z: parseFloat(firstCargoItem.dimension_z) * parseInt(firstCargoItem.quantity, 10),
  dimension_x: parseFloat(firstCargoItem.dimension_x) * parseInt(firstCargoItem.quantity, 10),
  payload_in_kg: parseFloat(firstCargoItem.payload_in_kg) * parseInt(firstCargoItem.quantity, 10),
  quantity: 1,
  groupAlias: 1,
  cargo_group_id: firstCargoItem.id,
  chargeable_weight: parseFloat(firstCargoItem.chargeable_weight) * parseInt(firstCargoItem.quantity, 10),
  hsCodes: firstCargoItem.hs_codes,
  hsText: firstCargoItem.hs_text,
  cargoType: { description: 'FOO_TYPE' },
  volume:
    parseFloat(firstCargoItem.dimension_y) *
    parseFloat(firstCargoItem.dimension_x) *
    parseFloat(firstCargoItem.dimension_z) /
    1000000 *
    parseInt(firstCargoItem.quantity, 10),
  items: [firstCargoItem]
}