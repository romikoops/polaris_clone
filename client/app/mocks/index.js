export * from './address'
export * from './cards'
export * from './cargoItems'
export * from './client'
export * from './consignee'
export * from './contacts'
export * from './containers'
export * from './currencies'
export * from './dimensions'
export * from './documents'
export * from './feeHash'
export * from './functions'
export * from './gMaps'
export * from './group'
export * from './hub'
export * from './match'
export * from './notifyees'
export * from './offer'
export * from './results'
export * from './routes'
export * from './schedules'
export * from './scope'
export * from './shipmentData'
export * from './shipments'
export * from './shipper'
export * from './tenant'
export * from './theme'
export * from './user'
