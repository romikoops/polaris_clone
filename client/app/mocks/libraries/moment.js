import moment from 'moment'

require('moment-timezone')

moment.tz.setDefault('Europe/Amsterdam')
