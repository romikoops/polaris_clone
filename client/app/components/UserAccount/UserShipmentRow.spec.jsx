import '../../mocks/libraries/moment'
import * as React from 'react'
import { shallow } from 'enzyme'
import { theme, identity, shipment } from '../../mocks'

import { UserShipmentRow } from './UserShipmentRow'

const editedShipment = {
  ...shipment,
  origin_hub_id: 'FOO',
  destination_hub_id: 'BAR',
  schedules_charges: {
    'FOO-BAR': {
      cargo: []
    }
  },
  schedule_set: [
    { hub_route_key: 'FOO-BAR' }
  ]
}

const hubs = {
  FOO: { data: { name: 'FOO_NAME' } },
  BAR: { data: { name: 'BAR_NAME' } }
}

const propsBase = {
  theme,
  user: {},
  handleSelect: identity,
  handleAction: identity,
  hubs,
  shipment: editedShipment
}

test('shallow render', () => {
  expect(shallow(<UserShipmentRow {...propsBase} />)).toMatchSnapshot()
})
