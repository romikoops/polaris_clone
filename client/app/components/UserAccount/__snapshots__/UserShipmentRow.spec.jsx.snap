// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props.shipment.schedule_set.length < 1 1`] = `
<div
  className="flex-100 layout-row pointy route_result"
  key="RANDOM_KEY_1"
>
  <div
    className="port_box flex-25 layout-row layout-wrap"
    onClick={[Function]}
  >
    <div
      className="flex-100 layout-row layout-align-center-center"
    >
      <h5
        className="flex-none no_m"
      >
         
        Booking Placed at: undefined
      </h5>
    </div>
    <div
      className="hub_half flex-100 layout-row layout-align-center-center"
    >
      <div
        className="flex-10 layout-row layout-align-center-center"
      >
        <i
          className="flex-none fa fa-map-marker map_marker"
        />
      </div>
      <div
        className="flex layout-row layout-align-center-center"
      >
        <h4
          className="flex-100 center letter_3"
        >
           
          FOO_NAME
           
        </h4>
      </div>
    </div>
    <div
      className="hub_mid_line flex-none layout-row layout-align-space-around-center layout-wrap"
    >
      <div
        className="flex-100 layout-row layout-align-center-end"
      >
        <i
          className="fa fa-ship"
        />
      </div>
      <div
        style={
          Object {
            "background": "linear-gradient(to right, transparent 70%, white 30%), linear-gradient(to right, #333, #fafafa)",
            "backgroundSize": "16px 2px, 100% 2px",
            "height": "2px",
            "marginTop": "6px",
            "width": "100%",
          }
        }
      />
    </div>
    <div
      className="hub_half flex-100 layout-row layout-align-center-center"
    >
      <div
        className="flex-10 layout-row layout-align-center-center"
      >
        <i
          className="flex-none fa fa-flag-o flag"
        />
      </div>
      <div
        className="flex layout-row layout-align-center-center"
      >
        <h4
          className="flex-100 center letter_3"
        >
           
          BAR_NAME
           
        </h4>
      </div>
    </div>
  </div>
  <div
    className="main_info flex layout-row layout-align-none-start layout-wrap"
    onClick={[Function]}
  >
    <div
      className="flex-100 layout-row layout-align-none-center"
    >
      <div
        className="flex-50 layout-row layout-align-start-center"
      >
        <h4
          className="flex-none no_m letter_3"
        >
           
          Ref: FOO_IMC_REFERENCE
        </h4>
      </div>
      <div
        className="flex-50 layout-row layout-align-end-center"
      >
        <h4
          className="flex-none letter_3 no_m"
        >
           
          0 X  
        </h4>
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-align-none-center"
    >
      <div
        className="user_info flex-50 layout-row layout-align-start-center"
      >
        <i
          className="flex-none fa fa-user flag"
          style={
            Object {
              "background": "-webkit-linear-gradient(left, undefined, undefined)",
            }
          }
        />
        <h4
          className="flex-none letter_3"
        >
           
          undefined undefined
           
        </h4>
      </div>
      <div
        className="flex-50 layout-row layout-align-end-center"
      >
        <h4
          className="flex-none letter_3 no_m"
        >
           
          Total: CNY 109
        </h4>
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-align-none-center"
    >
      <div
        className="flex-25 layout-wrap layout-row layout-align-center-center"
      >
        <div
          className="flex-100 layout-row"
        >
          <h4
            className="date_title"
          >
            Closing Date
          </h4>
        </div>
        <div
          className="flex-100 layout-row"
        >
          <p
            className="flex-none sched_elem"
          >
             
             
          </p>
          <p
            className="flex-none sched_elem"
          >
             
             
          </p>
        </div>
      </div>
      <div
        className="flex-25 layout-wrap layout-row layout-align-center-center"
      >
        <div
          className="flex-100 layout-row"
        >
          <h4
            className="date_title"
          >
             ETD
          </h4>
        </div>
        <div
          className="flex-100 layout-row"
        >
          <p
            className="flex-none sched_elem"
          >
             
            789
             
          </p>
          <p
            className="flex-none sched_elem"
          >
             
            789
             
          </p>
        </div>
      </div>
      <div
        className="flex-25 layout-wrap layout-row layout-align-center-center"
      >
        <div
          className="flex-100 layout-row"
        >
          <h4
            className="date_title"
          >
             ETA 
          </h4>
        </div>
        <div
          className="flex-100 layout-row"
        >
          <p
            className="flex-none sched_elem"
          >
             
             
          </p>
          <p
            className="flex-none sched_elem"
          >
             
             
          </p>
        </div>
      </div>
      <div
        className="flex-25 layout-wrap layout-row layout-align-center-center"
      >
        <div
          className="flex-100 layout-row"
        >
          <h4
            className="date_title"
          >
             Estimated Transit Time 
          </h4>
        </div>
        <div
          className="flex-100 layout-row"
        >
          <p
            className="flex-none sched_elem"
          >
             
              Days
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`shallow render 1`] = `
<div
  className="flex-100 layout-row pointy route_result"
  key="RANDOM_KEY_0"
>
  <div
    className="port_box flex-25 layout-row layout-wrap"
    onClick={[Function]}
  >
    <div
      className="flex-100 layout-row layout-align-center-center"
    >
      <h5
        className="flex-none no_m"
      >
         
        Booking Placed at: undefined
      </h5>
    </div>
    <div
      className="hub_half flex-100 layout-row layout-align-center-center"
    >
      <div
        className="flex-10 layout-row layout-align-center-center"
      >
        <i
          className="flex-none fa fa-map-marker map_marker"
        />
      </div>
      <div
        className="flex layout-row layout-align-center-center"
      >
        <h4
          className="flex-100 center letter_3"
        >
           
          FOO_NAME
           
        </h4>
      </div>
    </div>
    <div
      className="hub_mid_line flex-none layout-row layout-align-space-around-center layout-wrap"
    >
      <div
        className="flex-100 layout-row layout-align-center-end"
      >
        <i
          className="fa fa-ship"
        />
      </div>
      <div
        style={
          Object {
            "background": "linear-gradient(to right, transparent 70%, white 30%), linear-gradient(to right, #333, #fafafa)",
            "backgroundSize": "16px 2px, 100% 2px",
            "height": "2px",
            "marginTop": "6px",
            "width": "100%",
          }
        }
      />
    </div>
    <div
      className="hub_half flex-100 layout-row layout-align-center-center"
    >
      <div
        className="flex-10 layout-row layout-align-center-center"
      >
        <i
          className="flex-none fa fa-flag-o flag"
        />
      </div>
      <div
        className="flex layout-row layout-align-center-center"
      >
        <h4
          className="flex-100 center letter_3"
        >
           
          BAR_NAME
           
        </h4>
      </div>
    </div>
  </div>
  <div
    className="main_info flex layout-row layout-align-none-start layout-wrap"
    onClick={[Function]}
  >
    <div
      className="flex-100 layout-row layout-align-none-center"
    >
      <div
        className="flex-50 layout-row layout-align-start-center"
      >
        <h4
          className="flex-none no_m letter_3"
        >
           
          Ref: FOO_IMC_REFERENCE
        </h4>
      </div>
      <div
        className="flex-50 layout-row layout-align-end-center"
      >
        <h4
          className="flex-none letter_3 no_m"
        >
           
          0 X  
        </h4>
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-align-none-center"
    >
      <div
        className="user_info flex-50 layout-row layout-align-start-center"
      >
        <i
          className="flex-none fa fa-user flag"
          style={
            Object {
              "background": "-webkit-linear-gradient(left, undefined, undefined)",
            }
          }
        />
        <h4
          className="flex-none letter_3"
        >
           
          undefined undefined
           
        </h4>
      </div>
      <div
        className="flex-50 layout-row layout-align-end-center"
      >
        <h4
          className="flex-none letter_3 no_m"
        >
           
          Total: CNY 109
        </h4>
      </div>
    </div>
    <div
      className="flex-100 layout-row layout-align-none-center"
    >
      <div
        className="flex-25 layout-wrap layout-row layout-align-center-center"
      >
        <div
          className="flex-100 layout-row"
        >
          <h4
            className="date_title"
          >
            Closing Date
          </h4>
        </div>
        <div
          className="flex-100 layout-row"
        >
          <p
            className="flex-none sched_elem"
          >
             
             
          </p>
          <p
            className="flex-none sched_elem"
          >
             
             
          </p>
        </div>
      </div>
      <div
        className="flex-25 layout-wrap layout-row layout-align-center-center"
      >
        <div
          className="flex-100 layout-row"
        >
          <h4
            className="date_title"
          >
             ETD
          </h4>
        </div>
        <div
          className="flex-100 layout-row"
        >
          <p
            className="flex-none sched_elem"
          >
             
            789
             
          </p>
          <p
            className="flex-none sched_elem"
          >
             
            789
             
          </p>
        </div>
      </div>
      <div
        className="flex-25 layout-wrap layout-row layout-align-center-center"
      >
        <div
          className="flex-100 layout-row"
        >
          <h4
            className="date_title"
          >
             ETA 
          </h4>
        </div>
        <div
          className="flex-100 layout-row"
        >
          <p
            className="flex-none sched_elem"
          >
             
             
          </p>
          <p
            className="flex-none sched_elem"
          >
             
             
          </p>
        </div>
      </div>
      <div
        className="flex-25 layout-wrap layout-row layout-align-center-center"
      >
        <div
          className="flex-100 layout-row"
        >
          <h4
            className="date_title"
          >
             Estimated Transit Time 
          </h4>
        </div>
        <div
          className="flex-100 layout-row"
        >
          <p
            className="flex-none sched_elem"
          >
             
              Days
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
